cmd/soroban-rpc/lib/preflight/src/fees.rs

```diff
@@ -361,50 +355,63 @@ pub(crate) fn compute_bump_footprint_exp_transaction_data_and_min_fee(
     )
 }
 
-pub(crate) fn compute_restore_footprint_transaction_data_and_min_fee(
-    footprint: LedgerFootprint,
-    ledger_storage: &ledger_storage::LedgerStorage,
-    bucket_list_size: u64,
+fn compute_bump_footprint_rent_changes(
+    footprint: &LedgerFootprint,
+    ledger_storage: &LedgerStorage,
+    ledgers_to_expire: u32,
     current_ledger_seq: u32,
-) -> Result<(SorobanTransactionData, i64), Box<dyn error::Error>> {
-    let ConfigSettingEntry::StateExpiration(state_expiration) =
-        ledger_storage.get_configuration_setting(ConfigSettingId::StateExpiration)?
-        else {
-            return Err(
-                "get_fee_configuration(): unexpected config setting entry for StateExpiration key".into(),
-            );
-        };
+) -> Result<Vec<LedgerEntryRentChange>> {
     let mut rent_changes: Vec<LedgerEntryRentChange> = Vec::new();
-    for key in footprint.read_write.as_vec() {
-        let unmodified_entry = ledger_storage.get(key, true)?;
+    for key in (&footprint).read_only.as_vec() {
+        let unmodified_entry = ledger_storage
+            .get(key, false)
+            .with_context(|| format!("cannot get ledger entry with key {key:?}"))?;
```

Comment:
The key here ( pun intended ) is that we cannot bump the footprint of a key that we can't find on chain. The error that you've provided isn't incorrect, but doesn't deliver the "concept" that a key is expected to be on chain for it's expiration to be bumped.

Reply:
how would you rephrase it?

Reply:
OK, I slightly rephrased it.